File: DS/Queue.py
class Node:
    def __init__(self,info,next=None):
        self.info=info
        self.next=next

class Queue:
    def __init__(self):
        self.front=self.rear=None

    def isEmpty(self):
        return self.front==None
    
    def EnQueue(self,ele):
        n=Node(ele)
        if self.isEmpty():
            self.front=self.rear=n
            return 
        self.rear.next=n
        self.rear=n
        
    def DeQueue(self):
        if self.isEmpty():
            print("Queue is Empty ")
            return
        temp=self.front
        self.front=temp.next
        if self.front==None:
            self.rear=None    
        return temp

    def display(self):
        p=self.front
        while p!=None:
            print(p.info)
            p=p.next 
q=Queue()
q.EnQueue(10)
q.EnQueue(10)
q.EnQueue(100)
q.EnQueue(70)
print(q.display())
print("******************")
print(q.DeQueue())
print(q.DeQueue())
print(q.display())
                

File: DS/Queue.go
package main

import "fmt"

type Node struct{
    Info int
    Next  *Node
}

type Queue struct{
    Front *Node
    Rear *Node
}

func New()*Queue{
    q:=Queue{nil,nil}
    return &q
}

func(q * Queue) isEmpty()bool{
    return q.Front==nil
}


func(q * Queue) EnQueue(i int ){

    n:=Node{Info: i}
    if q.Front==nil{
        q.Front=&n
        q.Rear=&n
    }
    temp:=q.Rear
    temp.Next=&n
    q.Rear=&n

}

func(q * Queue) DeQueue() int {
    if q.isEmpty(){
        return -1
    }
    temp:=q.Front
    q.Front=temp.Next
    if q.Front==nil{
        q.Rear=nil
    }
    return temp.Info
}

func(q * Queue) Display()  {
    if q.isEmpty(){
        return
    }
    p:=q.Front
    for p!=nil{
        fmt.Println(p.Info)
        p=p.Next
    }
}

func main(){

    q:=New()
    q.EnQueue(10)
    q.EnQueue(10)
    q.EnQueue(100)
    q.EnQueue(70)
    q.Display()
    fmt.Println("******************")
    fmt.Println(q.DeQueue())
    fmt.Println(q.DeQueue())
    fmt.Println("******************")
    q.Display()

}
                

